Code/PgSQL/rdkit/CMakeLists.txt

```diff
@@ -0,0 +1,171 @@
+              rdkit_gist.c rdkit_io.c rxn_op.c sfp_op.c)
+  target_link_libraries(${EXTENSION} ${PostgreSQL_LIBRARIES})
+  if(WIN32)
+    target_link_libraries(${EXTENSION} postgres)
+  endif(WIN32)
+endif(APPLE)
+set(avalonRegress "")
+set(inchiRegress "")
+set(pgRDKitLibSuffix "")
+if(RDK_PGSQL_STATIC AND (NOT MSVC))
+  set(pgRDKitLibSuffix "_static")
+endif()
+set(pgRDKitLibList "")
+if(RDK_BUILD_AVALON_SUPPORT)
+  set(avalonRegress "avalon")
+  set(pgRDKitLibList "${pgRDKitLibList}AvalonLib;avalon_clib;")
+endif(RDK_BUILD_AVALON_SUPPORT)
+if(RDK_BUILD_INCHI_SUPPORT)
+  set(inchiRegress "inchi")
+  set(pgRDKitLibList "${pgRDKitLibList}RDInchiLib;Inchi;")
+endif(RDK_BUILD_INCHI_SUPPORT)
+set(pgRDKitLibList "${pgRDKitLibList}"
+    "MolHash;FMCS;ChemReactions;ChemTransforms;"
+    "FileParsers;SmilesParse;Fingerprints;Subgraphs;Descriptors;"
+    "PartialCharges;SubstructMatch;GraphMol;DataStructs;Depictor;"
+    "RDGeometryLib;RDGeneral")
+foreach(pgRDKitLib ${pgRDKitLibList})
+  set(pgRDKitLibs "${pgRDKitLibs}${pgRDKitLib}${pgRDKitLibSuffix};")
+endforeach()
+target_link_libraries(${EXTENSION}${EXTENSION_SUFFIX} ${pgRDKitLibs})
+if(RDK_BUILD_THREADSAFE_SSS)
+  target_link_libraries(${EXTENSION}${EXTENSION_SUFFIX} ${RDKit_THREAD_LIBS})
+endif(RDK_BUILD_THREADSAFE_SSS)
+
+if(WIN32 AND NOT UNIX)
+  set(PGREGRESS_BINARY "${PG_BINDIR}\\pg_regress.exe")
+  string(REGEX REPLACE "/" "\\\\" PG_EXTENSIONDIR ${PG_EXTENSIONDIR})
+  string(REGEX REPLACE "/" "\\\\" PGREGRESS_BINARY ${PGREGRESS_BINARY})
+else(WIN32 AND NOT UNIX)
+  run_pg_config (--pgxs PG_MAKEFILE)
+  get_filename_component(PG_MAKEFILESDIR ${PG_MAKEFILE} PATH)
+  set(PGREGRESS_BINARY "${PG_MAKEFILESDIR}/../test/regress/pg_regress")
+endif(WIN32 AND NOT UNIX)
+if(NOT EXISTS ${PGREGRESS_BINARY})
+  message(FATAL_ERROR "${PGREGRESS_BINARY} does not exist")
+endif()
+set(testPgSQLName "${CMAKE_CURRENT_BINARY_DIR}/pgsql_regress")
+set(installPgSQLName "${CMAKE_CURRENT_BINARY_DIR}/pgsql_install")
+set(PG_CURRENT_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/")
+set(PG_CURRENT_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/")
+if(WIN32 AND NOT UNIX)
+  set(testPgSQLCommand "")
+  set(testPgSQLName "${testPgSQLName}.bat")
+  set(installPgSQLName "${installPgSQLName}.bat")
+  set(installCopyCommand "copy /Y")
+  set(PG_RDKIT_LIB_SRC "${EXTENSION}.dll")
+  set(PG_RDKIT_LIB_DEST "${EXTENSION}.dll")
+  set(PGREGRESS_BINDIR_SWITCH "bindir")
+  string(REGEX REPLACE "/" "\\\\" PG_CURRENT_SOURCE_DIR ${PG_CURRENT_SOURCE_DIR})
+  string(REGEX REPLACE "/" "\\\\" PG_CURRENT_BINARY_DIR ${PG_CURRENT_BINARY_DIR})
+  string(REGEX REPLACE "/" "\\\\" PG_PKGLIBDIR ${PG_PKGLIBDIR})
+else(WIN32 AND NOT UNIX)
+  set(testPgSQLCommand "sh")
+  set(testPgSQLName "${testPgSQLName}.sh")
+  set(installPgSQLName "${installPgSQLName}.sh")
+  set(installCopyCommand "cp")
+  set(PG_RDKIT_LIB_SRC "${EXTENSION_PREFIX}${EXTENSION}.so")
+  set(PG_RDKIT_LIB_DEST "${EXTENSION}.so")
+  set(PGREGRESS_BINDIR_SWITCH "psqldir")

```

Comment:
Hi Greg,

OK, I am going to run a few tests tonight as soon as I get to my laptop.

Thanks, cheers
p.

On Thu, Mar 24, 2016 at 4:14 AM, Greg Landrum <notifications@github.com>
wrote:

> In Code/PgSQL/rdkit/CMakeLists.txt
> <https://github.com/rdkit/rdkit/pull/785#discussion_r57272958>:
>
> > +  set(installPgSQLName "${installPgSQLName}.bat")
> > +  set(installCopyCommand "copy /Y")
> > +  set(PG_RDKIT_LIB_SRC "${EXTENSION}.dll")
> > +  set(PG_RDKIT_LIB_DEST "${EXTENSION}.dll")
> > +  set(PGREGRESS_BINDIR_SWITCH "bindir")
> > +  string(REGEX REPLACE "/" "\\\\" PG_CURRENT_SOURCE_DIR ${PG_CURRENT_SOURCE_DIR})
> > +  string(REGEX REPLACE "/" "\\\\" PG_CURRENT_BINARY_DIR ${PG_CURRENT_BINARY_DIR})
> > +  string(REGEX REPLACE "/" "\\\\" PG_PKGLIBDIR ${PG_PKGLIBDIR})
> > +else(WIN32 AND NOT UNIX)
> > +  set(testPgSQLCommand "sh")
> > +  set(testPgSQLName "${testPgSQLName}.sh")
> > +  set(installPgSQLName "${installPgSQLName}.sh")
> > +  set(installCopyCommand "cp")
> > +  set(PG_RDKIT_LIB_SRC "${EXTENSION_PREFIX}${EXTENSION}.so")
> > +  set(PG_RDKIT_LIB_DEST "${EXTENSION}.so")
> > +  set(PGREGRESS_BINDIR_SWITCH "psqldir")
>
> This flag is unrecognized with postgresSQL 9.5:
>
> glandrum@Otter:/scratch/RDKit_git/build_new/Code/PgSQL/rdkit$ "/usr/local/pgsql/lib/pgxs/src/makefiles/../test/regress/pg_regress" --inputdir=sql --psqldir="/usr/local/pgsql/bin" rdkit-91 props btree molgist bfpgist-91 sfpgist slfpgist fps reaction avalon inchi
> /usr/local/pgsql/lib/pgxs/src/makefiles/../test/regress/pg_regress: unrecognized option '--psqldir=/usr/local/pgsql/bin'
>
> Try "pg_regress -h" for more information.
>
> but running without it works fine for me (on linux):
>
> glandrum@Otter:/scratch/RDKit_git/build_new/Code/PgSQL/rdkit$ "/usr/local/pgsql/lib/pgxs/src/makefiles/../test/regress/pg_regress" --inputdir=sql rdkit-91 props btree molgist bfpgist-91 sfpgist slfpgist fps reaction avalon inchi
> (using postmaster on Unix socket, default port)
> ============== dropping database "regression"         ==============
> NOTICE:  database "regression" does not exist, skipping
> DROP DATABASE
> ============== creating database "regression"         ==============
> CREATE DATABASE
> ALTER DATABASE
> ============== running regression test queries        ==============
> test rdkit-91                 ... ok
> test props                    ... ok
> test btree                    ... ok
> test molgist                  ... ok
> test bfpgist-91               ... ok
> test sfpgist                  ... ok
> test slfpgist                 ... ok
> test fps                      ... ok
> test reaction                 ... ok
> test avalon                   ... ok
> test inchi                    ... ok
>
> ======================
>  All 11 tests passed.
> ======================
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rdkit/rdkit/pull/785/files/8adfc6503a1cfd60697f6397a1fe99987f5b351c#r57272958>
>